pkg/controller/webhooks/postgresql.go

```diff
@@ -0,0 +1,199 @@
+package webhooks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/viper"
+	vshnv1 "github.com/vshn/appcat/apis/vshn/v1"
+	"github.com/vshn/appcat/pkg/common/quotas"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// This go generate and kube builder marker are only here for completeness sake.
+// Unfortunately the webhook generator is not very configurable yet and some
+// values cannot be changed. This will clash if/when more webhooks are generated.
+
+// Generate the webhook manifest from the marker comment
+// //go:generate go run -tags generate sigs.k8s.io/controller-tools/cmd/controller-gen webhook paths=./... output:dir=../../../config/controller/pg-webhooks
+
+// See https://book.kubebuilder.io/reference/markers/webhook for docs
+//+kubebuilder:webhook:verbs=create;update;delete,path=/validate-vshn-appcat-vshn-io-v1-vshnpostgresql,mutating=false,failurePolicy=fail,groups=vshn.appcat.vshn.io,resources=postgresqls,versions=v1,name=postgresql.vshn.appcat.vshn.io,sideEffects=None,admissionReviewVersions=v1
+
+var (
+	pgGK = schema.GroupKind{Group: "vshn.appcat.vshn.io", Kind: "VSHNPostgreSQL"}
+	pgGR = schema.GroupResource{Group: pgGK.Group, Resource: "vshnpostgresqls"}
+)
+
+var _ webhook.CustomValidator = &PostgreSQLWebhookHandler{}
+
+// PostgreSQLWebhookHandler handles all quota webhooks concerning postgresql by vshn.
+type PostgreSQLWebhookHandler struct {
+	client    client.Client
+	log       logr.Logger
+	withQuota bool
+}
+
+// SetupPostgreSQLWebhookHandlerWithManager registers the validation webhook with the manager.
+func SetupPostgreSQLWebhookHandlerWithManager(mgr ctrl.Manager, withQuota bool) error {
+	if !viper.IsSet("PLANS_NAMESPACE") && withQuota {
```

Comment:
shouldn't that be ensured by cmd/controller.go?

Reply:
good point.